src/Definition/ClassDefinition.php

```diff
@@ -110,6 +117,14 @@ public function compile(Context $context)
             return true;
         }
 
+        $context->getEventManager()->fire(
+            Event\StatementBeforeCompile::EVENT_NAME,
+            new Event\StatementBeforeCompile(
+                $this->statement,
```

Comment:
This should put `$this` instead of `Class_` Node, We are working with own Abstraction called as` Definition`s